### PR TITLE
[RFR] Add tests related to volume type RFE

### PR DESCRIPTION
**Pull request by jadhaj**
Purpose or Intent
=================
Added test_create_backup_of_volume_with_type
Added test_instance_attach_volume_with_type

{{ pytest: cfme/tests/openstack/cloud/test_instances.py::test_instance_attach_detach_volume_with_type cfme/tests/openstack/cloud/test_volume_backup.py::test_create_backup_of_volume_with_type }}

### CODE REVIEW

File: cfme/tests/openstack/cloud/test_instances.py [hunk 1]
```diff
@@ -236,3 +263,30 @@ def test_instance_attach_volume(new_instance, volume, appliance):
...
+def test_instance_attach_detach_volume_with_type(volume_with_type, new_instance, appliance):
+    initial_volume_count = new_instance.volume_count
+    new_instance.attach_volume(volume_with_type.name)
+    view = appliance.browser.create_view(navigator.get_class(new_instance, 'Details').VIEW)
+    view.flash.assert_success_message(
+        'Attaching Cloud Volume "{}" to {} finished'.format(volume_with_type.name, new_instance.name))
+
+    @wait_for_decorator(delay=10, timeout=300,
+                        message="Waiting for volume to be attached to instance")
+    def volume_attached_to_instance():
```
Comment (digitronik) [inline cfme/tests/openstack/cloud/test_instances.py]: Required:
I would like to shift this wait to `attach_volume` method with default argument `wait=False`.

`new_instance.attach_volume(volume_with_type.name, wait=True)` It should wait for volume attach. 
Similarly for `detach_volume` method.

File: cfme/cloud/instance/openstack.py [hunk 2]
```diff
@@ -64,8 +65,7 @@ def attach_volume(self, name, mountpoint=None):
 
     def detach_volume(self, name):
         view = navigate_to(self, 'DetachVolume')
-        view.form.volume.fill(name)
-        view.form.submit_button.click()
+        view.fill({'volume': name})
```
[the remark this answers is not in the document]
Comment (jadhaj) [reply cfme/cloud/instance/openstack.py]: yes, it was moved to DetachVolumeView under after_fill method
@jawatts  and @mshriver already know about that

[the remark this answers is not in the document]
Comment (jadhaj) [reply cfme/cloud/instance/openstack.py]: this workaround was introduced due to this bug:
https://bugzilla.redhat.com/show_bug.cgi?id=1647695

File: cfme/cloud/instance/openstack.py [hunk 3]
```diff
@@ -124,15 +124,22 @@ class form(View):  # noqa
...
-        submit_button = Button('Submit')
-        cancel_button = Button('Cancel')
+    volume = BootstrapSelect('volume_id')
+    submit_button = Button('Submit')
+    cancel_button = Button('Cancel')
 
     # Only the instance name is displayed, cannot confirm provider
     is_displayed = displayed_not_implemented
 
+    def after_fill(self, was_change):
```
[the remark this answers is not in the document]
Comment (jadhaj) [reply cfme/cloud/instance/openstack.py]: NO, As @mshriver explained to me, this is the way it should be implemented

File: cfme/cloud/instance/openstack.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jadhaj) [reply cfme/cloud/instance/openstack.py]: Regarding Wait for attach and detach the volume, they was part of the methods but they were moved outside for the assert success to work. When you do the wait which performs refresh inside the method the flash message will disappear despite creating the View again

File: cfme/tests/openstack/cloud/test_instances.py (hunk 1 above)
Comment (jadhaj) [reply cfme/tests/openstack/cloud/test_instances.py]: @digitronik Regarding Wait for attach and detach the volume, they was part of the methods but they were moved outside for the assert success to work. When you do the wait which performs refresh inside the method the flash message will disappear despite creating the View again

File: cfme/cloud/instance/openstack.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (digitronik) [reply cfme/cloud/instance/openstack.py]: As per BZ view distortion starting after `submit`. As per my opinion, we should handle this in `detach_volume` method with proper To-Do. Let's wait for @mshriver comment.

File: cfme/cloud/instance/openstack.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jadhaj) [reply cfme/cloud/instance/openstack.py]: Regarding 2:
There id no way right now to update volume type of a volume after it was crated

[the remark this answers is not in the document]
Comment (digitronik) [reply cfme/cloud/instance/openstack.py]: @jadhaj please check volume update method may be helpful to you.

[the remark this answers is not in the document]
Comment (jadhaj) [reply cfme/cloud/instance/openstack.py]: @digitronik  Yes I know it, but CFME UI does not allow changing volume type even manually

File: cfme/cloud/instance/openstack.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jawatts) [reply cfme/cloud/instance/openstack.py]: I agree with @digitronik here. The behavior we are working around here is not due to the submit button but the page redirection after the `Submit` button was clicked.

File: cfme/cloud/instance/openstack.py [hunk 4]
```diff
@@ -124,15 +124,22 @@ class form(View):  # noqa
...
+        # TODO: BZ1647695
```
Comment (jawatts) [inline cfme/cloud/instance/openstack.py]: Please add more information to this TODO like `Remove this method once BZ1647695 has been fixed`.

File: cfme/cloud/instance/openstack.py [hunk 5]
```diff
@@ -124,15 +124,22 @@ class form(View):  # noqa
...
+        if was_change:
```
Comment (jawatts) [inline cfme/cloud/instance/openstack.py]: Moving the `self.submit_button.click()` out of this, the `if` statement should be updated to:
```if was_change and BZ(BZ1647695, forced_streams=['5.10', 'upstream']).blocks```

That way when the bug is fixed, we no longer have to worry about this workaround.

File: cfme/cloud/instance/openstack.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jadhaj) [reply cfme/cloud/instance/openstack.py]: Got the point. thanks @jawatts  and @digitronik

**Pull request merged**